lib/data/schema.js

```diff
@@ -538,22 +538,15 @@ const injectPublicKey = (xml, keystr) => new Promise((pass, fail) => {
 const _versionSplicer = (replace) => (xml, insert) => new Promise((pass, fail) => {
   const stack = [];
   const parser = new hparser.Parser({
-    onattribute: (name, value) => {
+    onattribute: (name) => {
       if ((stripNamespacesFromPath(name) === 'version') && (stack.length) === 4 &&
         (stack[0] === 'html') && (stack[1] === 'head') && (stack[2] === 'model') &&
         (stack[3] === 'instance')) {
-        // okay, we have found the thing we are looking for.
-        // idx points at the position of the closing "
-        //
-        // TODO: we cheat here and reference the hp2 internal tokenizer index to find
-        // out where the attribute actually is. the parser startIndex and endIndex point
-        // at the whitespace preceding the tag until the tag is closed. obviously this is
-        // pretty bad but i don't see a more robust solution right now.
-        const idx = parser.tokenizer.index;
+        const result = replace
+          ? `${xml.slice(0, parser.startIndex)}version="${insert}"${xml.slice(parser.endIndex)}`
+          : `${xml.slice(0, parser.endIndex-1)}${insert}"${xml.slice(parser.endIndex)}`;
         parser.reset();
-        return replace
-          ? pass(`${xml.slice(0, idx - value.length)}${insert}${xml.slice(idx)}`)
-          : pass(`${xml.slice(0, idx)}${insert}${xml.slice(idx)}`);
+        return pass(result);
```

Comment:
latest version has fixed the bug around `startIndex` and `endIndex`, so we can use that.